### ISSUE TITLE
Custom validation error message not working - Backstage

### ISSUE THREAD

**Issue by Akshay-Durgade** [backstage/backstage#25936]
### 🔖 Need

I created a template for creating a repository, in which I have added a pattern for a particular field using the [Backstage Documentation](https://backstage.io/docs/features/software-templates/input-examples#custom-validation-error-message).

When the validation fails, a different error output is displayed than expected one.

This is the code of template.yaml
```yaml
apiVersion: scaffolder.backstage.io/v1beta3
kind: Template
# some metadata about the template itself
metadata:
  name: misc-config-Template
  title:  Misc Config Repo Template
  description: This template will be used for creating misc-config repos
  annotations:
    github.com/project-slug: backstage/backstage
spec:
  owner: xyz
  type: GitHub

  # these are the steps which are rendered in the frontend with the form input
    # these are the steps which are rendered in the frontend with the form input
  parameters:
    - title: Name of the catalog component
      required:
        - name
      properties:
        name:
          title: Name
          type: string
          description: Unique name of the component
          pattern: '^[a-z]{3}-deployment-config$'
          ui:autofocus: true
          ui:options:
            rows: 5
        owner:
          title: Name of the owner/organisation
          type: string
          description: Unique name of the component
          placeholder: "Name of the owner/organisation"
          ui:options:
          enum:
            - 'abc-def'
      errorMessage:
        properties:
          name: 'Input should be <xyz>-deployment-config'
          owner: 'Input should be firstName-lastName'
```

Expected output error message is:
`Input should be <xyz>-deployment-config`

Output error message displayed is:
`must match pattern '^[a-z]{3}-deployment-config$'`

### 🎉 Proposal

How do I solve this issue, even after following documentation?

### 👀 Have you spent some time to check if this RFC has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

**Comment by benjdlambert** [comment]
What version of Backstage are you running?

**Comment by benjdlambert** [comment]
I just realised that this is only available in the `-next` releases at the moment, it's due for release Aug 20th: https://github.com/backstage/backstage/pull/25624#issuecomment-2239332904

**Comment by Akshay-Durgade** [comment]
Thanks for the updates. Is there any option by which we can use the `-next` before it's release.

**Comment by awanlin** [comment]
Hi @Akshay-Durgade, if you run: `yarn backstage-cli versions:bump --release next` that will bump your Backstage instance to the latest `next` release.

**Comment by twanbeeren** [comment]
Is there by any means a chance that this still isn't resolved in 1.32?

For some reason my errormessage.name still doesn'tshow the new error message

My template:
```
  parameters:
    #Change parameters
    - title: Component information
      required:
        - name
        - description
        - owner
        - project
      properties:
        name:
          title: Name
          type: string
          description: Name for the service (-project is automatically appended)
          ui:field: EntityNamePicker
          ui:help: 'Maximum length can only be 20 characters'
          maxLength: 20
        description:
          title: Description
          type: string
          description: Description fit for the service
        owner:
          title: Owner
          type: string
          description: Owner team of the service
          ui:field: MyGroupsPicker
          ui:options:
            allowArbitraryValues: false
            catalogFilter:
              - kind: Group
                spec.type: team
      errorMessage:
        properties:
          name: 'The name should only use alphanumeric characters and spaces should be replaced with - (ex. test-api). Make sure it also does not surpass a length of 20 characters'
```

The error seem to still be the default:
![image](https://github.com/user-attachments/assets/62d30277-274a-41f7-981b-9abe1f61379a)

**Comment by NilashishC** [comment]
I'm using version `1.39.1` and I'm still facing this issue. Custom error messages are not displayed.
